lib/apple_sign_in/token.rb

```diff
@@ -76,5 +74,34 @@ def gen_private_key
       key = OpenSSL::PKey::EC.new(key) unless key.class == OpenSSL::PKey::EC
       key
     end
+
+    def client_urls
+      {
+        site: APPLE_AUD,
+        authorize_url: '/auth/authorize',
+        token_url: '/auth/token'
+      }
+    end
+
+    def reponse_hash(access_token)
+      token_hash = { access_token: access_token.token }
+
+      expired = access_token.expired?
+      if expired
+        token_hash[:expires_at] = access_token.expires_at
```

Comment:
@vitogit setting this is necessary for some purpose?

Reply:
I think is useful, to return the access_token, refresh_token and expired status.

Reply:
Ok, but why is it useful returning an expired date after it expired?

Reply:
The useful thing is to save the expired date, so you know when it's gonna expire. But I think the code should be rearrange to refresh the token before (so it's update the expired date) and then save the expired_date.  Also `expired?` and `expires` . Some tokens don't expires so `expires` is false (this is a oauth2 thing)
```
    access_token.refresh! if access_token.expired?

    token_hash = { 'access_token' => access_token.token }
    token_hash['refresh_token'] = access_token.refresh_token if access_token.expires? && access_token.refresh_token
    token_hash['expires_at'] = access_token.expires_at if access_token.expires?
```

Reply:
My code is the same:
in line 21:
```  access_token.refresh! if access_token.expired? ```
in line 23:
``` reponse_hash(access_token)```
that method do
```
      token_hash = { access_token: access_token.token }

      expired = access_token.expired?
      if expired
        token_hash[:expires_at] = access_token.expires_at
        refresh_token = access_token.refresh_token
        token_hash[:refresh_token] = refresh_token if refresh_token
      end
```
the mistake that i see is: I put  ```expired = access_token.expired?``` and should be ```expires = access_token.expires?```

Reply:
Yes you are right. That's looks good.